src/JsonAsynchronousNodeKit/LargeJsonParser.cs

```diff
@@ -0,0 +1,281 @@
+            }
+
+            ParserUtils.AssertTokenType(this.stream, ref reader, JsonTokenType.PropertyName);
+            var propertyName = reader.GetString() ?? throw new NotImplementedException();
+            ParserUtils.Read(this.stream, ref this.buffer, ref reader);
+
+            ParserStateResult resultState;
+            if (this.handlers.ContainsKey(propertyName))
+            {
+                resultState = this.HandleExplicitProperty(ref reader, propertyName);
+            }
+            else
+            {
+                resultState = this.HandleExtraProperty(ref reader, propertyName);
+            }
+
+            // yield return is a special case where we need to leave the reader in the same state as we found it.
+            if (!resultState.YieldReturn)
+            {
+                ParserUtils.GetMoreBytesFromStream(this.stream, ref this.buffer, ref reader);
+
+                this.isFinalBlock = reader.IsFinalBlock;
+                this.readerState = reader.CurrentState;
+            }
+
+            this.previousResultState = resultState;
+
+            return resultState;
+        }
+        catch (JsonException ex)
+        {
+            throw new ParserException($"Error parsing json at position {this.stream.Position}", ex);
+        }
+    }
+
+    private ParserStateResult HandleExplicitProperty(ref Utf8JsonReader reader, string propertyName)
+    {
+        var handler = this.handlers![propertyName];
+
+        object? result;
+        switch (handler.Type)
+        {
+            case ParameterType.String:
+                ParserUtils.AssertTokenType(this.stream, ref reader, JsonTokenType.String);
+                result = reader.GetString();
+                break;
+            case ParameterType.Skip:
+                ParserUtils.SkipProperty(this.stream, ref this.buffer, ref reader);
+                result = null;
+                break;
+            case ParameterType.Int:
+                ParserUtils.AssertTokenType(this.stream, ref reader, JsonTokenType.Number);
+                var i = reader.GetInt32();
+                result = i;
+                break;
+            case ParameterType.Object:
+                var objType = handler.GetType().GetGenericArguments()[0];
+                result = this.GetObject(objType, ref reader);
+                break;
+            case ParameterType.Array:
+                var aryType = handler.GetType().GetGenericArguments()[0];
+                result = this.ParseArray(ref reader, aryType);
+                break;
+            default:
+                throw new InvalidOperationException($"Unknown {nameof(ParameterType)}: {handler.Type}");
+        }
+
+        return new ParserStateResult(propertyName, result, ExplicitField: true, YieldReturn: handler.Type == ParameterType.Array);
+    }
+
+    private ParserStateResult HandleExtraProperty(ref Utf8JsonReader reader, string propertyName)
+    {
+        // We skip objects and arrays to avoid having to parse them.
```

Comment:
I'm getting a bit lost here. It says that objects and arrays are skipped, but lines 190 and 191 calls `ParseArray` and `ParseObject`. Do you mean we skip parsing to specific object/enumerable, but we still have to parse to `JsonObject` and `JsonArray` in order to traverse them?

Reply:
If I understand it correctly, maybe it would be more clear?
```suggestion
        // We skip objects and arrays to avoid having to deserialize them.
```

Reply:
Good catch. I had originally been going to do ParserUtils.SkipProperty for "extra" array parameters, but I ended up deciding it made more sense to make people explicitly skip the properties they didn't want since otherwise it's difficult to explain why we don't return extra arrays but DO return extra objects. That being the case I think the solution is to just remove this outdated comment entirely.